modules/messages/account_parser.go

```diff
@@ -29,323 +19,48 @@ func MessageNotSupported(msg sdk.Msg) error {
+	for _, event := range tx.Events {
+		for _, attribute := range event.Attributes {
+			if strings.Contains(attribute.Value, "/") {
+				continue
 			}
-
-			switch {
-			case strings.Contains(index.TypeUrl, "bank"):
-				addresses, err := BankMessagesParser(c, executedMsg)
-				if err != nil {
-					return nil, MessageNotSupported(executedMsg)
-				}
-				addresses = append(addresses, msg.Grantee)
-				return addresses, nil
-			case strings.Contains(index.TypeUrl, "crisis"):
-				addresses, err := CrisisMessagesParser(c, executedMsg)
-				if err != nil {
-					return nil, MessageNotSupported(executedMsg)
-				}
-				addresses = append(addresses, msg.Grantee)
-				return addresses, nil
-			case strings.Contains(index.TypeUrl, "distribution"):
-				addresses, err := DistributionMessagesParser(c, executedMsg)
-				if err != nil {
-					return nil, MessageNotSupported(executedMsg)
-				}
-				addresses = append(addresses, msg.Grantee)
-				return addresses, nil
-			case strings.Contains(index.TypeUrl, "evidence"):
-				addresses, err := EvidenceMessagesParser(c, executedMsg)
-				if err != nil {
-					return nil, MessageNotSupported(executedMsg)
-				}
-				addresses = append(addresses, msg.Grantee)
-				return addresses, nil
-			case strings.Contains(index.TypeUrl, "gov.v1beta1"):
-				addresses, err := GovV1Beta1MessagesParser(c, executedMsg)
-				if err != nil {
-					return nil, MessageNotSupported(executedMsg)
-				}
-				addresses = append(addresses, msg.Grantee)
-				return addresses, nil
-			case strings.Contains(index.TypeUrl, "gov.v1."):
-				addresses, err := GovV1MessageParser(c, executedMsg)
-				if err != nil {
-					return nil, MessageNotSupported(executedMsg)
-				}
-				addresses = append(addresses, msg.Grantee)
-				return addresses, nil
-			case strings.Contains(index.TypeUrl, "ibc"):
-				addresses, err := IBCTransferMessagesParser(c, executedMsg)
-				if err != nil {
-					return nil, MessageNotSupported(executedMsg)
-				}
-				addresses = append(addresses, msg.Grantee)
-				return addresses, nil
-			case strings.Contains(index.TypeUrl, "slashing"):
-				addresses, err := SlashingMessagesParser(c, executedMsg)
-				if err != nil {
-					return nil, MessageNotSupported(executedMsg)
-				}
-				addresses = append(addresses, msg.Grantee)
-				return addresses, nil
-			case strings.Contains(index.TypeUrl, "staking"):
-				addresses, err := StakingMessagesParser(c, executedMsg)
-				if err != nil {
-					return nil, MessageNotSupported(executedMsg)
-				}
-				addresses = append(addresses, msg.Grantee)
-				return addresses, nil
-			default:
-				addresses, err := DefaultMessagesParser(c, executedMsg)
-				if err != nil {
-					return nil, MessageNotSupported(executedMsg)
-				}
-				addresses = append(addresses, msg.Grantee)
-				return addresses, nil
+			if strings.Contains(attribute.Value, chainPrefix) {
+				allAddressess = append(allAddressess, attribute.Value)
```

Comment:
This won't work. If the attribute is, for example `desmos-subspace-1` then it would be inserted as an address. 

Instead, we should make sure the address is a valid format first. This can be done using `sdk.AccAddressFromBech32` or the other methods (if we want to include validator addresses as well). Then, if the error of the parsing is not nill, we include the address inside the list of parsed addresses. Otherwise we skip it.